src/core/packages/saved-objects/api-server-internal/src/lib/search/search_dsl/sorting_params.ts

```diff
@@ -45,22 +45,46 @@ export function getSortingParams(
 
   if (types.length > 1) {
     const rootField = getProperty(mappings, sortField);
-    if (!rootField) {
-      throw Boom.badRequest(
-        `Unable to sort multiple types by field ${sortField}, not a root property`
-      );
-    }
+    if (rootField) {
+      return {
+        sort: [
+          {
+            [sortField]: {
+              order: sortOrder,
+              unmapped_type: rootField.type,
+            },
+          },
+        ],
+      };
+    } else {
+      // rootField not found, create a runtime field that merges the sort fields from all types
+      const mergedFieldName = `merged_${sortField}`;
+      const scriptLines = types.map((t, idx) => {
+        const fieldName = `${t}.${sortField}`;
+        return idx === 0
+          ? `if (doc['${fieldName}'].size() != 0) { emit(doc['${fieldName}'].value); }`
+          : `else if (doc['${fieldName}'].size() != 0) { emit(doc['${fieldName}'].value); }`;
+      });
+      const scriptSource = scriptLines.join(' ');
 
-    return {
-      sort: [
-        {
-          [sortField]: {
-            order: sortOrder,
-            unmapped_type: rootField.type,
+      return {
+        runtime_mappings: {
+          [mergedFieldName]: {
+            type: 'keyword',
```

Comment:
is it safe to assume that this is always a `keyword`? I wonder if we want to sort by a numeric field present in multiple types.

Reply:
i added checks around that